### DISCUSSION TITLE
missing a function called "GetFragments" in scATAC integration vignette

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#632] [Q&A] [answered]
Hi Tim,
I am working on the integration of two scATAC datasets created by 10x.
I did some QC for each of them (https://satijalab.org/signac/articles/pbmc_vignette.html) and then follow the integration vignette (https://satijalab.org/signac/0.2/articles/integration.html) 
When I was creating the intersecting.regions with the separators both as c(":", "-") , I was notified that (":", "-") was not used though I did mention "sep" when I created the ChromatinAssay:

```
chrom_assay.6798 <- CreateChromatinAssay(
  counts = counts.6798,
  sep = c(":", "-"),
  fragments = 'fragments.tsv.gz',
  min.cells = 10,
  min.features = 200
)
```

 So, I commented the "sep" and went ahead. 
```

intersecting.regions <- GetIntersectingFeatures(
  object.1 = pbmc.6798,
  object.2 = pbmc.6800,
  # sep.1 = c(":", "-"),
  # sep.2 = c(":", "-")
)
```

here is what intersecting.regions look like:

` intersecting.regions
`
[[1]]
    [1]      1      2      4      5      6      7      8      9     10     11
   [11]     12     13     14     15     16     17     19     21     25     27
   [21]     28     29     30     31     32     33     34     35     36     38
   [31]     39     40     42     43     44     45     46     47     48     49
   [41]     50     51     52     54     56     57     58     59     61     63
   [51]     64     65     66     67     68     69     70     72     73     74
   [61]     75     77     78     79     80     81     82     86     87     88

But when it came to FeatureMatrix part which is:

```
pbmc.6800_peaks_pbmc.6798 <- FeatureMatrix(
  fragments = GetFragments(object = pbmc.6798, assay = 'peaks'),
  features = StringToGRanges(peaks.use),
  cells = colnames(pbmc.6798)
)
```

I was told that "_Error in GetFragments(object = pbmc.6798, assay = "peaks") : 
  could not find function "GetFragments"_" though I have loaded Signac and Seurat. 

Can you give me some advice on how to fix these problems?
Thanks

**Comment by timoast** [comment, marked as answer]
You're looking at a very old version of the integration vignette, take note of the Signac version number shown in the website you're looking at. Here is the current vignette: https://satijalab.org/signac/articles/integrate_atac.html

**Reply by pengxin2019** [reply]
Thanks Tim for sharing the newest vignette with me. 
I remember in one of the previous discussions under Signac,  you suggested someone to use Harmony in terms of scATAC integration. I just had a quick look of the newest vignette you shared. It seems like Harmony is not suggested any more, right?

Also, I have two scATAC datasets now. But I am going to add more datasets which means I would have more than two datasets in total (they are all independent biological replicates). How is the newsest integration vignette with more than two datasets created by 10x now? Have you detected any potential problem or heard something from other user?

Thanks

**Reply by pengxin2019** [reply]
what is improved in this new vignette compared to the previous one? Batch effect correction?

**Reply by timoast** [reply]
The new version should perform much better and scale to large datasets, and enable integration of many (>2) datasets. You can certainly still run Harmony, we have just removed the demonstration of Harmony since the Seurat integration methods now work well for scATAC-seq data.

**Comment by pengxin2019** [comment]
Hi Tim,
I am doing integration of two scATAC datasets produced by 10x  following the pipeline here: https://satijalab.org/signac/articles/integrate_atac.html#integration-1

But it is reporting an error when it comes to the FindIntegrationAnchors method of integration:
![Screen Shot 2021-05-20 at 7 29 14 PM](https://user-images.githubusercontent.com/54915282/119064753-ab50bb00-b9a1-11eb-897b-855d021e360d.png)
The document in the vignette is saying that we should use "rlsi" but document of the FindIntegrationAnchors method is saying that we should choose  "cca" or "rpca". I am confused.

Can you give me some suggestions?
Thanks

**Reply by timoast** [reply]
This requires the develop version of Seurat (as stated in bold in the vignette) https://satijalab.org/seurat/articles/install.html#install-the-development-version-of-seurat-1

**Reply by pengxin2019** [reply]
Hi Tim,
In terms of the VlnPlot plot created in the Find differentially accessible peaks between clusters here: https://satijalab.org/signac/articles/pbmc_vignette.html

![Screen Shot 2021-06-07 at 7 36 44 AM](https://user-images.githubusercontent.com/54915282/121017640-1f8dab80-c763-11eb-82c3-4527993e155f.png)
what is the creteria to define that this gene is differentially expressed in these two clusters? I can see that the p value is extremely small which is reasonalble to say they are different. But are you using the average expresssion value in both groups? It seems like (from the plot) their average are kind of close.


Also, I am wondering what is the meaning of the vertical bar in the CD14 Mono group in this plot. why we have this bar in this group?

By the way, I see that you are using the "FindMarkers" function to find markers between two groups. Do you think it performs better than FindAllMarker which can give us the markers for each cell type compared to all other cell types?

Thanks

**Reply by timoast** [reply]
Since this is completely unrelated to this discussion topic, please open a new discussion. Also see the documentation for `VlnPlot()` and `FindMarkers()`, where you should find answers to most of these questions
